src/provider/mod.rs

```diff
@@ -155,7 +157,17 @@ impl Provider for StackableProvider {
 
     async fn initialize_pod_state(&self, pod: &Pod) -> anyhow::Result<Self::PodState> {
         let service_name = pod.name();
-        let service_uid = String::from(pod.as_kube_pod().metadata.uid.as_ref().unwrap());
+
+        // Extract uid from pod object, if this fails we return an error -
+        // this should not happen, as all objects we get from Kubernetes should have
+        // a uid set!
+        let service_uid = if let Some(uid) = pod.as_kube_pod().metadata.uid.as_ref() {
```

Comment:
This can be changed to 

```
        let service_uid = pod
            .as_kube_pod()
            .metadata
            .uid
            .as_ref()
            .ok_or(anyhow::Error::new(IllegalKubeObject {
                field_name: "uid".to_string(),
            }))?
            .clone();
```
Not sure if it's better...